src/snowcli/output/printing.py

```diff
@@ -25,99 +28,85 @@ def default(self, o):
-        cursor: SnowflakeCursor, format: Optional[OutputFormat] = None
+        cls, cursor: SnowflakeCursor, format_: Optional[OutputFormat] = None
     ) -> OutputData:
-        check_if_is_cursor(cursor)
-        return OutputData(cursor, format)
-
-    @staticmethod
-    def from_string(message: str, format: Optional[OutputFormat] = None) -> OutputData:
-        check_if_is_string(message)
-        return OutputData(message, format)
-
-    @staticmethod
-    def from_list(data: List[Dict], format: Optional[OutputFormat] = None):
-        check_if_is_list(data)
-        return OutputData(data, format)
-
-    def add_cursor(self, cursor: SnowflakeCursor) -> OutputData:
-        check_if_is_cursor(cursor)
-        self.output.append(cursor)
-        self.counter += 1
-        return self
-
-    def add_string(self, message: str) -> OutputData:
-        check_if_is_string(message)
-        self.output.append(message)
-        self.counter += 1
-        return self
-
-    def add_list(self, data: List[Dict]):
-        check_if_is_list(data)
-        self.output.append(data)
-        self.counter += 1
-        return self
+        """Converts Snowflake cursor to stream of data"""
+        if not isinstance(cursor, SnowflakeCursor):
+            raise OutputDataTypeError(type(cursor), SnowflakeCursor)
+        return OutputData(stream=_get_data_from_cursor(cursor), format_=format_)
+
+    @classmethod
+    def from_string(
+        cls, message: str, format_: Optional[OutputFormat] = None
+    ) -> OutputData:
+        """Coverts string to stream of data"""
+        if not isinstance(message, str):
+            raise OutputDataTypeError(type(message), str)
+        return cls(stream=({"result": message} for _ in range(1)), format_=format_)
+
+    @classmethod
+    def from_list(
+        cls, data: List[Union[Dict, OutputData]], format_: Optional[OutputFormat] = None
+    ):
+        """Converts list to stream of data."""
+        if not isinstance(data, list) or (
+            len(data) > 0 and not isinstance(data[0], (dict, OutputData))
+        ):
+            raise OutputDataTypeError(type(data), List[Union[Dict, OutputData]])
+        return cls(stream=(item for item in data), format_=format_)
 
     @property
     def format(self):
         if not self._format:
             self._format = _get_format_type()
         return self._format
 
-    def get_data(self) -> Union[str, Dict, Iterable[List[Dict]]]:
-        for output in self.output:
-            if isinstance(output, SnowflakeCursor):
-                yield _get_data_from_cursor(output)
-            elif isinstance(output, str):
-                if self.format != OutputFormat.JSON:
-                    yield output
-                else:
-                    yield {"result": output}
-            elif isinstance(output, list):
-                yield output
-
-    def size(self) -> int:
-        return self.counter
-
+    def empty(self):
+        return self._stream is None
```

Comment:
What with case when stream has 0 items?

Reply:
Can stream have 0 items and not be None?

Reply:
You are the python expert 😄